### DISCUSSION TITLE
Caching data

### DISCUSSION THREAD

**Discussion by hellhorse123** [TanStack/query#6935] [Q&A] [answered]
In this React ts code I get stageData:
{"status":"created"}

Then i invoke function, which update status to processing on server side. In Postman i see, that status is changed, But in my code i still get status: created and only hard reload helps:

```
 const {
    data: stageData,
    isLoading: stageLoading,
    error: stageError,
  } = useQuery<IStageData, Error>(
    'str',
    () =>
      StagesService.getStage(
        projectId || "",
        experimentId || "",
        stageId || ""
      ),
    {
      onSuccess: (data) => {
        setStageStatus(data.status);
        onSuccess?.(data);
      },
      staleTime: 0,
      cacheTime: 0,
    }
  );
```

(If I choose disable cache in devTools console, error is gone)

**Comment by TkDodo** [comment]
have you invalidated the query ?

also, you might have browser caching turned on, which prevents you from getting updated data. You should see the network request showing "(from browser cache)" in the devtools.

**Reply by hellhorse123** [reply]
I found out that it turns out that react-query does not work through cache control

**Reply by TkDodo** [reply, marked as answer]
Of course not - it doesn't do data fetching. Promises don't know about any headers
